pytorch_fairseq_translation.md

```diff
@@ -0,0 +1,122 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+title: Transformer (NMT)
+summary: Transformer models for English-French and English-German translation.
+category: researchers
+image: fairseq_logo.png
+author: Facebook AI Research (fairseq Team)
+tags: [nlp]
+github-link: https://github.com/pytorch/fairseq/
+accelerator: cuda-optional
+order: 10
+---
+
+
+### Model Description
+
+The Transformer, introduced in the paper [Attention Is All You Need][1], is a
+powerful sequence-to-sequence modeling architecture capable of producing
+state-of-the-art neural machine translation (NMT) systems.
+
+Recently, the fairseq team has explored large-scale semi-supervised training of
+Transformers using back-translated data, further improving translation quality
+over the original model. More details can be found in [this blog post][2].
+
+
+### Requirements
+
+We require a few additional Python dependencies for preprocessing:
+
+```bash
+pip install regex requests sacremoses sentencepiece subword_nmt
+```
+
+
+### English-to-French Translation
+
+To translate from English to French using the model from the paper [Scaling
+Neural Machine Translation][3]:
+
+```python
+import torch
+en2fr = torch.hub.load(
+  'pytorch/fairseq',
+  'transformer',
+  model_name_or_path='transformer.wmt14.en-fr',
+  data_name_or_path='.',
+  tokenizer='moses',
+  aggressive_dash_splits=True,
+  bpe='subword_nmt',
+)
+print(en2fr.generate('Hello world!'))
+# Bonjour à tous !
+```
+
+
+### English-to-German Translation
+
+Semi-supervised training with back-translation is an effective way of improving
+translation systems. In the paper [Understanding Back-Translation at Scale][4],
+we back-translate over 200 million German sentences to use as additional
+training data. An ensemble of these models was the winning submission to the
+[WMT'18 English-German news translation competition][5].
+
+To translate from English to German using one of the models from the winning
+ensemble (note: the archive includes the full ensemble and is 16GB uncompressed,
+so will take some time to download and extract):
+
+```python
+import torch
+en2de = torch.hub.load(
+  'pytorch/fairseq',
+  'transformer',
+  model_name_or_path='transformer.wmt18.en-de',
+  checkpoint_file='wmt18.model1.pt',
+  data_name_or_path='.',
+  tokenizer='moses',
+  aggressive_dash_splits=True,
+  bpe='subword_nmt',
+)
+len(en2de.models)
+# 1
+print(en2de.generate('Pytorch Hub is a pre-trained model repository designed to facilitate research reproducibility.'))
+# Pytorch Hub ist ein vortrainiertes Modelldepot, das die Reproduzierbarkeit von Forschungsergebnissen erleichtern soll.
+```
+
+We can also translate using the whole ensemble, but this requires a lot of GPU
```

Comment:
what is a "whole ensemble", clarify?
Also, warn them that the "whole ensemble" is like several gigabytes of download?